pkg/image/containerdregistry/registry_test.go

```diff
@@ -0,0 +1,41 @@
+package containerdregistry
+
+import (
+	"archive/tar"
+	"testing"
+)
+
+func Test_fromFilter(t *testing.T) {
+	type destination struct {
+		dest   string
+		match  bool
+		header *tar.Header
+	}
+	var destinations = []destination{
+		{
+			dest:   "/",
+			header: &tar.Header{Name: "root"},
+			match:  true,
+		},
+		{
+			dest:   "my-db.db",
+			header: &tar.Header{Name: "my-db.db"},
+			match:  true,
+		},
+		{
+			dest:   "my-db.db",
+			header: &tar.Header{Name: "rocks-db.db"},
+			match:  false,
+		},
+		// TODO directory test cases
```

Comment:
Is this still needed?

Reply:
Added some directory tests to this (and refactored some).